R/ergm_mplecov.R

```diff
@@ -0,0 +1,100 @@
+#  File R/ergm_mplecov.R in package ergm, part of the Statnet suite
+#  of packages for network analysis, https://statnet.org .
+#
+#  This software is distributed under the GPL-3 license.  It is free,
+#  open source, and has the attribution requirements (GPL Section 7) at
+#  https://statnet.org/attribution
+#
+#  Copyright 2003-2020 Statnet Commons
+#######################################################################
+#' Approximate MPLE standard errors in dyad-dependent models
+#' 
+#' Function to approximate the MPLE covariance matrix in a dyad dependence model 
+#' using the Godambe matrix as described in Schmid and Hunter (2020) or by parametric bootstrap
+#' as described by Schmid and Desmarais (2017).
+#' 
+#' @param pl An \code{\link{ergm.pl}} object.
+#' @param nw response network.
+#' @param fd An \code{\link{rlebdm}} with informative dyads.
+#' @param m the model, as returned by \code{\link{ergm_model}}
+#' @param init a vector a vector of initial theta coefficients
+#' @param theta.mple the MPLE of a given model
+#' @param invHess the inverse Hessian matrix obtained from glm()
+#' @param control a list of MCMC related parameters; recognized
+#'   components include: samplesize : the number of networks to sample
+#'   Clist.miss : see 'Clist.miss' above; some of the code uses this
+#'   Clist.miss,
+#' @param verbose whether this and the C routines should be verbose (T
+#'   or F); default=FALSE
+#' @return \code{ergm_mplecov} returns a list either
+#'   containing a Godambe covariance matrix or a diagonal matrix with bootstrap variances.
+#' 
+#' @references Schmid CS and Desmarais BA (2017). "Exponential random graph 
+#' models with big networks: Maximum pseudolikelihood estimation and the parametric bootstrap" 
+#' _IEEE International Conference on Big Data (Big Data)_, pp. 116-121.
+#' 
+#' Schmid CS and Hunter DR (2020).  "Accounting for Model Misspecification When Using 
+#' Pseudolikelihood for ERGMs" _Working Paper_.
+#' @noRd
+ergm_mplecov <- function(pl,nw, fd, m, init=init, theta.mple, invHess,  control=NULL,
+                         verbose=FALSE){
+  
+  # get sample size from control.ergm
+  R <- control$MPLE.covariance.samplesize
+  
+  # Simulate R networks
+  sim.mple <- simulate(m, nsim=R, coef=theta.mple, basis=nw, control=control.simulate.formula(MCMC.burnin=5000, MCMC.interval=3000))
```

Comment:
Are you sure this is correct? It looks like it would ignore any sample space constraints (including the dyad-independent ones, which the MPLE does recognise). Also, burnin and interval should be control parameters somewhere.

Reply:
That's a good point! I will fix it.